guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,750 @@
+"""
+Let's make sure our augmentations look how we expect them to:
+"""
+visualize_dataset(train_ds, bounding_box_format="xywh")
+
+"""
+Great!  We now have a bounding box friendly data augmentation pipeline.
+Let's format our evaluation dataset to match.  Instead of using
+`JitteredResize`, let's use the deterministic `keras_cv.layers.Resizing()`
+layer.
+"""
+
+inference_resizing = keras_cv.layers.Resizing(
+    640, 640, bounding_box_format="xywh", pad_to_aspect_ratio=True
+)
+eval_ds = eval_ds.map(inference_resizing, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Due to the fact that the resize operation differs between the train dataset,
+which uses `JitteredResize()` to resize images, and the inference dataset which
+uses `layers.Resizing(pad_to_aspect_ratio=True)`, it is good practice to
+visualize both datasets:
+"""
+
+visualize_dataset(eval_ds, bounding_box_format="xywh")
+
+"""
+Finally, let's unpackage our inputs from the preprocessing dictionary, and
+prepare to feed the inputs into our model.  In order to be TPU compatible,
+bounding box Tensors need to be `Dense` instead of `Ragged`.  If training on
+GPU, you can omit the `bounding_box.to_dense()` call.  If ommitted,
+the KerasCV RetinaNet
+label encoder will automatically correctly encode Ragged training targets.
+"""
+
+
+def dict_to_tuple(inputs):
+    return inputs["images"], bounding_box.to_dense(
+        inputs["bounding_boxes"], max_boxes=32
+    )
+
+
+train_ds = train_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+Our data pipeline is now complete!
+We can now move on to model creation and training.
+
+### Model creation
+
+We'll use the KerasCV API to construct a RetinaNet model.
+In this tutorial we use a pretrained ResNet50 backbone, initializing the
+weights to weights produced by training a classification model on the imagenet
+dataset.
+
+KerasCV makes it easy to construct a `RetinaNet` with any of the KerasCV
+backbones.  Simply use one of the presets for the architecture you'd like!
+
+For example:
+"""
+
+model = keras_cv.models.RetinaNet.from_preset(
+    "resnet50_v2_imagenet",
+    # number of classes to be used in box classification
+    num_classes=len(class_mapping) + 1,
+    # For more info on supported bounding box formats, visit
+    # https://keras.io/api/keras_cv/bounding_box/
+    bounding_box_format="xywh",
+)
+
+"""
+That is all it takes to construct a KerasCV RetinaNet.  The RetinaNet accepts
+tuples of dense image Tensors and bounding box dictionaries to `fit()` and
+`train_on_batch()`
+
+This matches what we have constructed in our input pipeline above.
+"""
+
+"""
+### Optimizer
+
+Next, let's construct a learning rate schedule and optimizer for our model.
+"""
+base_lr = 0.005 * (BATCH_SIZE / 16)
```

Comment:
Might be good to offer some more explanation and visualize the LR schedule. This is an advanced topic.